File: sdk/python/aura/types/bridge.py
"""Type definitions for Bridge module."""

from dataclasses import dataclass
from typing import Optional, List
from enum import Enum
from datetime import datetime


class BridgeStatus(Enum):
    """Bridge transfer status."""

    PENDING = "pending"
    CONFIRMED = "confirmed"
    COMPLETED = "completed"
    FAILED = "failed"
    REFUNDED = "refunded"


class ChainType(Enum):
    """Supported chain types."""

    ETHEREUM = "ethereum"
    BINANCE = "binance"
    POLYGON = "polygon"
    COSMOS = "cosmos"


@dataclass
class BridgeTransferParams:
    """Parameters for initiating bridge transfer."""

    source_chain: str
    destination_chain: str
    token: str
    amount: str
    recipient: str
    memo: Optional[str] = None
    timeout_height: Optional[int] = None
    timeout_timestamp: Optional[int] = None


@dataclass
class BridgeTransfer:
    """Bridge transfer information."""

    transfer_id: str
    source_chain: str
    destination_chain: str
    sender: str
    recipient: str
    token: str
    amount: str
    status: BridgeStatus
    created_at: datetime
    updated_at: datetime
    transaction_hash: Optional[str] = None
    destination_hash: Optional[str] = None
    proof: Optional[str] = None
    error_message: Optional[str] = None


@dataclass
class BridgeParams:
    """Bridge module parameters."""

    enabled: bool
    supported_chains: List[str]
    min_transfer_amount: str
    max_transfer_amount: str
    transfer_fee: str
    timeout_duration: int
    confirmation_blocks: int
    relayer_addresses: List[str]


@dataclass
class BridgeBalance:
    """Bridge balance information."""

    chain: str
    token: str
    locked_amount: str
    available_amount: str
    total_supply: str


@dataclass
class BridgeProof:
    """Bridge transfer proof."""

    transfer_id: str
    merkle_root: str
    merkle_proof: List[str]
    signatures: List[str]
    validators: List[str]


@dataclass
class RelayerInfo:
    """Bridge relayer information."""

    address: str
    chains: List[str]
    active: bool
    total_relayed: int
    success_rate: float
    last_relay_time: Optional[datetime] = None

File: sdk/javascript/src/types/bridge.ts
import { Coin } from '@cosmjs/stargate';

/**
 * Bridge transfer status enum
 */
export enum BridgeTransferStatus {
  PENDING = 0,
  COMPLETED = 1,
  FAILED = 2,
  REFUNDED = 3,
}

/**
 * Bridge transfer information
 */
export interface BridgeTransfer {
  id: string;
  sender: string;
  recipient: string;
  amount: Coin;
  sourceChain: string;
  targetChain: string;
  status: BridgeTransferStatus;
  createdAt: Date;
  completedAt?: Date;
  proof?: string;
  txHash?: string;
  error?: string;
}

/**
 * Parameters for initiating a bridge transfer
 */
export interface InitiateBridgeParams {
  sender: string;
  recipient: string;
  amount: string;
  denom: string;
  targetChain: string;
  timeout?: number;
  memo?: string;
}

/**
 * Parameters for completing a bridge transfer
 */
export interface CompleteBridgeParams {
  transferId: string;
  proof: string;
  height: number;
  signatures: string[];
}

/**
 * Bridge parameters
 */
export interface BridgeParams {
  minTransferAmount: string;
  maxTransferAmount: string;
  supportedChains: string[];
  bridgeFee: string;
  confirmationsRequired: number;
  timeout: number;
  enabled: boolean;
}

/**
 * Bridge security configuration
 */
export interface BridgeSecurity {
  merkleRoot: string;
  validators: string[];
  requiredSignatures: number;
  lastUpdateHeight: number;
}

/**
 * Bridge statistics
 */
export interface BridgeStats {
  totalTransfers: number;
  totalVolume: string;
  activeTransfers: number;
  completedTransfers: number;
  failedTransfers: number;
}
